packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -33,6 +35,16 @@ export interface DataSourceSrv {
    * Reloads the DataSourceSrv
    */
   reload(): void;
+
+  /**
+   * Registers a runtime data source that can be used from scene applications
+   * Make sure your data source uid is unique
+   */
+  registerRuntimeDataSource(entry: RuntimeDataSourceRegistration): void;
+}
+
+export interface RuntimeDataSourceRegistration {
+  dataSource: RuntimeDataSource;
```

Comment:
It seems counter intuitive that this is basically on the same level as any other data source but this has to be a `RuntimeDataSource` type. Like I assume `RuntimeDataSource` is sort of a helper so user don't have to recreate all the meta info all the time but seems weird to make it a requirement. Or is the `RuntimeDataSource-` prefix in the name something we want to make required?